.clang-tidy

```diff
@@ -1,5 +1,5 @@
 ---
-Checks:          'clang-diagnostic-*,clang-analyzer-*,-*,performance-*,-performance-inefficient-string-concatenation,mpi-*,modernize-*,-modernize-pass-by-value,-modernize-use-trailing-return-type,-modernize-avoid-c-arrays,-modernize-use-nodiscard,-modernize-concat-nested-namespaces,readability-*,-readability-braces-around-statements,-readability-named-parameter,-readability-magic-numbers,-readability-uppercase-literal-suffix,-readability-identifier-length,-readability-function-cognitive-complexity,-readability-redundant-string-cstr'
+Checks:          'clang-diagnostic-*,clang-analyzer-*,-*,performance-*,-performance-noexcept-swap,-performance-inefficient-string-concatenation,mpi-*,modernize-*,-modernize-pass-by-value,-modernize-use-trailing-return-type,-modernize-avoid-c-arrays,-modernize-use-nodiscard,-modernize-concat-nested-namespaces,readability-*,-readability-braces-around-statements,-readability-named-parameter,-readability-magic-numbers,-readability-uppercase-literal-suffix,-readability-identifier-length,-readability-function-cognitive-complexity,-readability-redundant-string-cstr,-readability-inconsistent-declaration-parameter-name'
```

Comment:
Why were `noexcept-swap` and `inconsistent-declaration-parameter-name` disabled? With the latter one, does it mean that there is no need to do the change in `ArborX_DetailDistributedTreeImpl.hpp`?

Reply:
Removing this checks produced no warnings (at least, for `Serial` build).

Reply:
> Reenabling these checks produced no warnings (at least, for Serial build).

Did you try a recent version of `clang-tidy`? I was running with `17.0.6`.

> Why were noexcept-swap and inconsistent-declaration-parameter-name disabled? 

Because they are noisy and don't add much value. `noexcept-swap` complains if our `swap` implementation is not `noexcept` but we probably don't care. `inconsistent-declaration-parameter-name` suggests a lot of changes that we could make if we care about naming our parameters consistently but it would be better to create another pull request for that.

> With the latter one, does it mean that there is no need to do the change in ArborX_DetailDistributedTreeImpl.hpp?

Yes, it seems I forgot to remove that one.

Reply:
> Did you try a recent version of `clang-tidy`? I was running with `17.0.6`.

Was running with 14.

> `noexcept-swap` complains if our `swap` implementation is not `noexcept` but we probably don't care. ``

Why don't we instead make our `swap` `noexcept`?

> `inconsistent-declaration-parameter-name` suggests a lot of changes that we could make if we care about naming our parameters consistently but it would be better to create another pull request for that.

Weird. The option is present in 14, so I should have seen a bunch of warnings. But I don't. It may be a good idea to at least take a look at it.

Reply:
> Why don't we instead make our `swap` `noexcept`?

Because that would be a lie. Our current interface looks like
```C++
template <class T>
KOKKOS_FUNCTION constexpr void
swap(T &a, T &b) noexcept(std::is_nothrow_move_constructible_v<T>
                              &&std::is_nothrow_move_assignable_v<T>);
```
so it's `noexcept` if it can be already. I would rather disable this very specific `check` than change the implementation.

> Weird. The option is present in 14, so I should have seen a bunch of warnings. But I don't. It may be a good idea to at least take a look at it.

If it doesn't show up in the CI (which it probably doesn't if you don't see it), it would be pretty hard to enforce (unless we update to a newer version).